### PR TITLE
Add bulk (CSV) import/export for more object types

### PR DESCRIPTION
**Pull request by paravoid**
### Fixes: #1347

This adds bulk (CSV) import & export for regions, rack groups, manufacturers and device types, which can be useful when e.g. importing from a legacy system, as described in #1347. 

Mostly copy/paste from the bulk/CSV forms of other object types and tested locally with a few fields for each (but not all), so a careful code review would be appropriate and appreciated!

### CODE REVIEW

File: netbox/dcim/forms.py [hunk 1]
```diff
@@ -70,6 +70,28 @@ class Meta:
         fields = ['parent', 'name', 'slug']
 
 
+class RegionCSVForm(forms.ModelForm):
+    parent = forms.ModelChoiceField(
```
Comment (jeremystretch) [inline netbox/dcim/forms.py]: This should probably be a `TreeNodeChoiceField` from `mptt.forms`.

File: netbox/dcim/forms.py [hunk 2]
```diff
@@ -396,6 +451,38 @@ class Meta:
...
+    manufacturer = forms.ModelChoiceField(
+        queryset=Manufacturer.objects.all(),
+        required=True,
+        to_field_name='name',
+        help_text='Manufacturer name',
+        error_messages={
+            'invalid_choice': 'Manufacturer not found.',
+        }
+    )
+
```
Comment (jeremystretch) [inline netbox/dcim/forms.py]: Can omit blank lines between model fields

File: netbox/dcim/forms.py [hunk 3]
```diff
@@ -396,6 +451,38 @@ class Meta:
...
+    subdevice_role = CSVChoiceField(
+        choices=SUBDEVICE_ROLE_CHOICES,
+        required=False,
+        help_text='Parent/child status'
+    )
+
+    interface_ordering = CSVChoiceField(
```
Comment (jeremystretch) [inline netbox/dcim/forms.py]: Can add `required=False` here as the field has a default value.

File: netbox/dcim/models.py [hunk 4]
```diff
@@ -52,6 +56,13 @@ def __str__(self):
     def get_absolute_url(self):
         return "{}?region={}".format(reverse('dcim:site_list'), self.slug)
 
+    def to_csv(self):
+        return csv_format([
+            self.name,
+            self.slug,
+            self.parent if self.parent else None,
```
Comment (jeremystretch) [inline netbox/dcim/models.py]: Should probably be `self.parent.name`

File: netbox/dcim/models.py [hunk 5]
```diff
@@ -500,6 +537,21 @@ def __init__(self, *args, **kwargs):
     def get_absolute_url(self):
         return reverse('dcim:devicetype', args=[self.pk])
 
+    def to_csv(self):
+        return csv_format([
+            self.manufacturer,
```
Comment (jeremystretch) [inline netbox/dcim/models.py]: `self.manufacturer.name` just to be explicit

File: netbox/dcim/models.py [hunk 6]
```diff
@@ -500,6 +537,21 @@ def __init__(self, *args, **kwargs):
...
+            self.model,
+            self.slug,
+            self.part_number,
+            self.u_height,
+            self.is_full_depth,
+            self.is_console_server,
+            self.is_pdu,
+            self.is_network_device,
+            self.get_subdevice_role_display() if self.subdevice_role else None,
+            self.get_interface_ordering_display() if self.interface_ordering else None,
```
Comment (jeremystretch) [inline netbox/dcim/models.py]: Just `self.get_interface_ordering_display()`: `interface_ordering` is a required field

**Review by jeremystretch — CHANGES_REQUESTED** [review]
Looks good! Just a couple minor tweaks and we can merge this. Thanks!

File: netbox/dcim/forms.py (hunk 1 above)
Comment (paravoid) [inline netbox/dcim/forms.py]: Not sure -- in my setup where I have a region "Europe", another one "US", with children "US East" and "US West", with this change it shows as:
```
Name of parent region
Choices: Europe, US, --- US East, etc. 
```
…while before this change it shows up as:
```
 Name of parent region
Choices: Europe, US, US East, etc. 
```

Comment (jeremystretch) [inline netbox/dcim/forms.py]: Ah, good point.

**Review by jeremystretch — APPROVED** [review]

**Comment by paravoid** [comment]
This was approved but not merged ­-- was that intentional? Anything I can do?

**Comment by jeremystretch** [comment]
@paravoid Could you resolve the conflicts in the base template please? Sorry, I had made some style changes. Other than that I've just been waiting to get to a point where I can implement the rest of the imports.

**Pull request merged**